File: about_cnv/select_diff_CNV.pl
use Getopt::Long；
GetOptions(
	"gtf=s" => \$gtf,
	"cnr=s" => \$cnr,
	"log=i" => \$log
);

open IN,$gtf,or die $!;
while(<IN>)
{
	chomp;
	next if (/##/);
        @a = split/\t/,$_;
        if ($a[2]=~/gene/ and $a[8]=~/gene_name "(\S+)";/){$gene = $1;$cc{$gene}{$a[0]}{$a[3]} = $a[4];}
}
open CNR,$cnr,or die $!;
while(<CNR>)
{
	chomp;
	
}

File: about_cnv/plot_CNV/select_diff_CNV.py
import linecache
import argparse
import numpy as np
import pprint
from itertools import islice 
import re
parser = argparse.ArgumentParser(description='manaul of this script')
parser.add_argument('--log',type = int,default = None)
parser.add_argument('--cnr',type = str,default = None)
parser.add_argument('--gtf',type = str,default = None)
args =parser.parse_args()

def addtwodimdict(thedict, key_a, key_b, val): 
  if key_a in adic:  
    thedict[key_a].update({key_b: val})
  else:
    thedict.update({key_a:{key_b: val}})

chrom = {}
chrom_s = {}
chrom_e = {}
dictc = {}
gg = {}

gtf = open(args.gtf).readlines()
for i in gtf:
  k = i.split('\t')
  if (len(k) >=9):
    if (k[2] == 'gene'):
      m = re.search('gene_name "(.+)";  l',k[8])
      gene = m.group(1)
      chrom[gene]=k[0]
      chrom_s[gene]=k[3]
      chrom_e[gene]=k[4]

f = open(args.cnr)
for i in islice(f, 1, None):
  v = float(i.split('\t')[4])
  mmm = 0 - args.log
  if (v > args.log or v < mmm):
    chro = i.split('\t')[0]
    star = i.split('\t')[1]
    #end = i.split('\t')[2]
    gg[chro] = star

for gene in chrom.keys():
  if chrom[gene] in gg.keys():
    if (gg[chrom[gene]] > chrom_s[gene] and gg[chrom[gene]] < chrom_e[gene]):
      print gene
